File: src/main/kotlin/customfit/ai/kotlinclient/client/AllFlagsListener.kt
package customfit.ai.kotlinclient.client

/**
 * An interface for objects that will be notified when any feature flag changes.
 * Register instances of this with [CFClient.registerAllFlagsListener].
 */
interface AllFlagsListener {
    /**
     * Called when any feature flag changes.
     *
     * @param flagMap a map of all current feature flags
     */
    fun onFlagsChange(flagMap: Map<String, Any>)
} 

File: customfit-flutter-client-sdk/lib/src/client/listener/all_flags_listener.dart
/// Listener interface for receiving all feature flag changes.
abstract class AllFlagsListener {
  /// Called when any feature flag changes.
  ///
  /// [oldFlags] is a map of all previous feature flags and their values.
  /// [newFlags] is a map of all current feature flags and their values.
  void onAllFlagsChanged(
      Map<String, dynamic> oldFlags, Map<String, dynamic> newFlags);
}
